src/Twig/Extension/AuthRuntime.php

```diff
@@ -191,6 +191,54 @@ public function getProviders($guid = null)
     }
 
     /**
+     * Return an array of registered OAuth providers for an account with
+     * lst steen and last updated dates
+     *
+     * @param string $guid
+     *
+     * @return array
+     */
+    public function getProvidersDates($guid = null)
+    {
+        $providers = [];
+        if ($guid === null) {
+            $auth = $this->session->getAuthorisation();
+
+            if ($auth === null) {
+              return $providers;
+            }
+            $guid = $auth->getGuid();
+        }
+
+        $providerEntities = $this->records->getProvisionsByGuid($guid);
+        if ($providerEntities === false) {
+            return $providers;
+        }
+
+        /** @var Storage\Entity\Provider $providerEntity */
+        $providers['lastseen'] = null;
+        $providers['lastupdate'] = null;
```

Comment:
Yeah, my problem is the inconsistency of the keys and datatypes.
```
$providers = [
    0 => [ 
        'name' => 'google',
        'lastseen' => 'some date',
        'lastupdate' => 'some date'
    ],
    1 => [ 
        'name' => 'github',
        'lastseen' => 'some later date',
        'lastupdate' => 'some later date'
    ]
];
```
would be fine,
```
$providers = [
    'lastseen' => 'some date',
    'lastupdate' => 'some date'
];
```
would also be fine, but mixing them seems like asking for trouble. Any time you loop it you need to know specifically which keys to exclude or have logic for both strings and arrays. There is already a way to get this data for the providers, what this feature/fix is supposed to do is to get a way to get it for the whole user, right? In that case this should be sufficent, right?:
```
$providers = [
    'lastseen' => 'some date',
    'lastupdate' => 'some date'
];
```